genienlp/evaluate_file.py

```diff
@@ -0,0 +1,188 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import json
+import logging
+import os
+from collections import defaultdict
+from pprint import pformat
+
+from .metrics import calculate_and_reduce_metrics
+from .models.base import ValidationOutput
+from .tasks.registry import get_tasks
+from .util import set_seed
+
+logger = logging.getLogger(__name__)
+
+
+def parse_argv(parser):
+    parser.add_argument(
+        '--pred_file',
+        required=True,
+        type=str,
+        help='Name of dataset to run prediction for; will be ignored if --evaluate is test',
+    )
+    parser.add_argument('--tasks', dest='task_names', nargs='+', required=True, help='task names for prediction')
+    parser.add_argument('--extra_metrics', nargs='+', default=[], help='include these additional metrics in reported results')
+    parser.add_argument(
+        '--devices',
+        default=None,
+        nargs='+',
+        type=int,
+        help='a list of devices that can be used for prediction. By default, all devices will be used.',
+    )
+    parser.add_argument('--seed', default=123, type=int, help='Random seed.')
+    parser.add_argument('--overwrite', action='store_true', help='whether to overwrite previously written predictions')
+    parser.add_argument('--silent', action='store_true', help='whether to print predictions to stdout')
+
+    parser.add_argument('--eval_dir', type=str, required=False, help='use this directory to store eval results')
+    parser.add_argument('--subsample', default=20000000, type=int, help='subsample the eval/test datasets')
+
+    parser.add_argument(
+        '--pred_languages',
+        type=str,
+        nargs='+',
+        dest='pred_src_languages',
+        default=['en'],
+        help='Specify dataset source languages used during prediction for multilingual tasks',
+    )
+    parser.add_argument(
+        '--pred_tgt_languages',
+        type=str,
+        nargs='+',
+        default=['en'],
+        help='Specify dataset target languages used during prediction for multilingual tasks',
+    )
+
+    parser.add_argument(
+        '--main_metric_only', action='store_true', help='If True, we only calculate the deca score metric for each task.'
```

Comment:
Add a check that raises `ValueError` if `main_metric_only` and `extra_metrics` are both specified. The same in `predict.py`.